src/prefect/cache_policies.py

```diff
@@ -21,6 +37,36 @@ def from_cache_key_fn(
         """
         return CacheKeyFnPolicy(cache_key_fn=cache_key_fn)
 
+    def configure(
+        self,
+        storage: Union["WritableFileSystem", str, Path, None] = None,
+        lock_manager: Optional["LockManager"] = None,
+        isolation_level: Union[
+            Literal["READ_COMMITTED", "SERIALIZABLE"], "IsolationLevel", None
+        ] = None,
+    ) -> Self:
+        """
+        Configure the cache policy with the given storage, locks, and isolation level.
+
+        Args:
+            storage: The storage to use for the cache policy. If not provided,
+                the current storage will be used.
+            locks: The lock manager to use for the cache policy. If not provided,
+                the current lock manager will be used.
+            isolation_level: The isolation level to use for the cache policy. If not provided,
+                the current isolation level will be used.
+
+        Returns:
+            A new cache policy with the given storage, locks, and isolation level.
+        """
+        if storage is not None:
+            self.storage = storage
+        if lock_manager is not None:
+            self.lock_manager = lock_manager
+        if isolation_level is not None:
+            self.isolation_level = isolation_level
+        return self
```

Comment:
commented elsewhere but your note about mutating our pre-instantiated policies was a good callout; I agree with your original implementation that we shouldn't mutate those and should instead start this off with a deepcopy